diff --git a/Library/Homebrew/env_config.rb b/Library/Homebrew/env_config.rb
index 83ec07682a..4cac73b2e5 100644
--- a/Library/Homebrew/env_config.rb
+++ b/Library/Homebrew/env_config.rb
@@ -707,6 +707,10 @@ module Homebrew
         default:     true,
         hidden:      true,
       },
+      HOMEBREW_SHUT_UP_ABOUT_UNSUPPORTED_OS:     {
+        description: "If set, do not warn that Homebrew does not support this macOS version or architecture.",
+        boolean:     :set,
+      },
       HOMEBREW_SIMULATE_MACOS_ON_LINUX:          {
         description: "If set, running Homebrew on Linux will simulate certain macOS code paths. This is useful " \
                      "when auditing macOS formulae while on Linux.",
diff --git a/Library/Homebrew/extend/os/mac/diagnostic.rb b/Library/Homebrew/extend/os/mac/diagnostic.rb
index 5c697f8fa1..4689fae9ac 100644
--- a/Library/Homebrew/extend/os/mac/diagnostic.rb
+++ b/Library/Homebrew/extend/os/mac/diagnostic.rb
@@ -129,6 +129,7 @@ module OS
         sig { returns(T.nilable(::Homebrew::Diagnostic::Finding)) }
         def check_for_unsupported_macos
           return if Homebrew::EnvConfig.developer?
+          return if Homebrew::EnvConfig.shut_up_about_unsupported_os?
           return if ENV["HOMEBREW_INTEGRATION_TEST"]
 
           tier = 2
