--- upstream/entr.rb
+++ Formula/entr.rb
@@ -1,14 +1,14 @@
 class Entr < Formula
   desc "Run arbitrary commands when files change"
   homepage "https://eradman.com/entrproject/"
-  url "https://eradman.com/entrproject/code/entr-5.8.tar.gz"
+  url "https://github.com/eradman/entr/archive/refs/tags/5.8.tar.gz"
   sha256 "dc9a2bdc556b2be900c1d8cdf432de26492de5af3ffade000d4bfd97f3122bfb"
   license "ISC"
   head "https://github.com/eradman/entr.git", branch: "master"
 
   livecheck do
-    url "https://eradman.com/entrproject/code/"
-    regex(/href=.*?entr[._-]v?(\d+(?:\.\d+)+)\.t/i)
+    url :head
+    regex(/^(\d+(?:\.\d+)+)$/i)
   end
 
   bottle do
